rules/microsoft_rules/microsoft365_external_sharing.py

```diff
@@ -0,0 +1,44 @@
+import re
+from fnmatch import fnmatch
+
+from panther_base_helpers import m365_alert_context
+
+email_regex = re.compile(r"([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+")
+
+ALLOWED_DOMAINS = ["mycompany.com", "alloweddomain.com"]
```

Comment:
I think it makes sense to  include an example value that includes `allowedoutsidedomain.com`

Reply:
The "Allowed User" test is testing for users allowed outside domain